WcaOnRails/app/webpacker/lib/hooks/useQueryParams.js

```diff
@@ -0,0 +1,30 @@
+import { useEffect, useState } from 'react';
+
+function getQueryParamsFromBrowserUrl() {
+  const entries = (new URLSearchParams(window.location.search)).entries();
+  return entries.reduce((accumulator, currentValue) => {
+    const [key, value] = currentValue;
+    accumulator[key] = value;
+    return accumulator;
+  }, {});
+}
+
+export default function useQueryParams() {
+  const [queryParams, setQueryParams] = useState(getQueryParamsFromBrowserUrl());
+
+  useEffect(() => {
+    const searchParams = new URLSearchParams(queryParams).toString();
+    if (window.location.search !== searchParams) {
```

Comment:
`!==` checks for referential equality. Since you are creating a `new URLSearchParams` object directly above, I feel like this comparison always yields `true` because they can by defintion never be the same.